TinyPG/Templates/C#/Scanner.cs

```diff
@@ -94,6 +94,8 @@ public Token LookAhead(params TokenType[] expectedtokens)
             Token tok = null;
             List<TokenType> scantokens;
 
+            string lowerInput = Input.ToLower();
+
```

Comment:
``Input`` is the entire file's content, right?
Does ``Input`` change at all?  It looks like it never does after ``Init()``.  But this routine, ``LookAhead()``, is getting called over and over and over.  Therefore you're applying the ``ToLower()`` again and again on the whole file content each time it looks for just one more token.  I'm going to try changing it to only apply ToLower() once during Init(), not again and again, and see if it works that way.